zarr/core.py

```diff
@@ -1731,11 +1750,16 @@ def _chunk_setitem_nosync(self, chunk_coords, chunk_selection, value, fields=Non
     def _chunk_key(self, chunk_coords):
         return self._key_prefix + '.'.join(map(str, chunk_coords))
 
-    def _decode_chunk(self, cdata):
+    def _decode_chunk(self, cdata, start=None, nitems=None, expected_shape=None):    
 
         # decompress
         if self._compressor:
-            chunk = self._compressor.decode(cdata)
+            # only decode requested items
+            if (all([x is not None for x in [start, nitems]])
+                and self._compressor.codec_id == 'blosc'):
```

Comment:
```suggestion
                and self._compressor.codec_id == 'blosc') and hasattr(self._compressor, 'decode_partial'):
```

And/or you may want to modify `.travis.yml` to install numcodec from master.